locksmithctl/daemon.go

```diff
@@ -116,6 +117,52 @@ func expBackoff(interval time.Duration) time.Duration {
 	return interval
 }
 
+// getPeriodicAndSleep get the REBOOT_WINDOW_START, REBOOT_WINDOW_LENGTH
+// and check if everything is ok and the get then Periodic. Check if
+// it is outside of the reboot window and If yes then sleep.
+func getPeriodicAndSleep() error {
+	var period *timeutil.Periodic
+
+	startw := os.Getenv("LOCKSMITHD_REBOOT_WINDOW_START")
+	if startw == "" {
+		startw = os.Getenv("REBOOT_WINDOW_START")
+	}
+
+	lengthw := os.Getenv("LOCKSMITHD_REBOOT_WINDOW_LENGTH")
+	if lengthw == "" {
+		lengthw = os.Getenv("REBOOT_WINDOW_LENGTH")
+	}
+
+	if (startw == "") != (lengthw == "") {
+		return errors.New("Either both or neither $REBOOT_WINDOW_START and $REBOOT_WINDOW_LENGTH must be set")
+	}
+
+	if startw != "" && lengthw != "" {
+		p, err := timeutil.ParsePeriodic(startw, lengthw)
+		if err != nil {
+			return err
```

Comment:
not for this PR but for your personal knowledge: a good practice for the error handling is to `wrap` the error to bring more context on this error in case you have multiple nested errors.
ref: https://blog.golang.org/go1.13-errors#TOC_3.3.
```suggestion
			return fmt.Errorf("unable to parse periodic: %w", err)
```